syntax="proto2";

import "netsim/acousticstoolbox/environment.proto";

//Messages for range-dependent Sound Velocity Profile extraction

package SVP;
 
message Request
{ 
required int32 request_id = 1;
required double utc_time = 2;
required double source_lat = 3;
required double source_lon = 4;
required double receiver_lat = 5;
required double receiver_lon = 6;  
required double delta_range = 7;
required double delta_depth = 8;
}

message Response
{
required int32 request_id = 1;
repeated bellhop.protobuf.Environment.WaterColumn water_column = 2;
}

message BathyRequest
{
required int32 request_id = 1;
repeated double latitude = 2;
repeated double longitude = 3;
}

message BathyResponse
{
required int32 request_id = 1;
repeated double bathy = 2;
}

message EOFCoefficients
{
repeated EOFWeight eof_weight = 1;
message EOFWeight
  {
    required int32 num = 1;
    required double value = 2;
  }
}
  
message EOFInternalWave
{
required double period = 1;
required double wavelength = 2;
required double heading = 3;
repeated EOFAmplitudeFactor eof_wave = 5;
message EOFAmplitudeFactor
  {
    optional int32 num = 1 ;
    optional double factor = 2 ;
  }
}  


